Feature/younghyun add filter, refactoring

younghyunlee22 opened the pull request:
* add 'rate' filter
* refactor handleSliderChange function
* consolidate repeated slider logic into a reusable FilterSlider component

younghyunlee22 committed:
add rate filter

younghyunlee22 committed:
Refactoring - handleSliderChange function

younghyunlee22 committed:
Refactor: Componentize FilterSlider

younghyunlee22 committed:
feat: Add filters state management for sliders in ResultFilters component

- Implemented Redux state management for filters in the ResultFilters component.
- Defined the SET_FILTERS action in the reducer to handle filter updates.
- Adjusted handleSliderChange logic to utilize the updated filters state from Redux.

hunjoonrhee reviewed (approved):
looks good to me! 👍

younghyunlee22 committed:
chore: action type constant

younghyunlee22 committed:
chore: eslint

younghyunlee22 committed:
fix: Apply trim to search inputs in both general and detailed searches

MayHyeyeonKim commented on src/app/components/Header.tsx:
Header의 검색에서 Search키를 클릭하면 trim된 데이타가 잘 날라가는데 Enter키 사용하면 여전히 trim이 안된 채로 날라가고 있어요.
Enter키 감지를 위해 handleKeyPress가 있던데 `event.preventDefault();`를 넣어보시겠어요?

MayHyeyeonKim replied:
(하쥐만 이 event.preventDefault(); 솔루션일 수 있는지 사실 모르겠어요 ㅋㅋ 찾아봐야해요)

MayHyeyeonKim replied:
setTimeout(() => setSearchTerm(''), 0); 일케하니 되네요!

MayHyeyeonKim replied:
```typescript
  const handleKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
    if (event.key === 'Enter') {
      event.preventDefault();
      const trimmedSearchTerm = searchTerm.trim();
      if (trimmedSearchTerm === '') {
        alert('Please enter a search term.');
        return;
      }
      const searchCondition = encodeURIComponent(JSON.stringify({ searchTerm: trimmedSearchTerm, page: 1, pageSize: 20 }));
      router.push(`/search/result?searchCondition=${searchCondition}`);
      dispatch(
        getBooksSearchRequest({
          page: initialPage,
          pageSize: initialPageSize,
          searchTerm: trimmedSearchTerm,
        }),
      );

      // 상태 초기화를 지연시켜서 최신 입력값 반영한 후에 초기화하기~
      setTimeout(() => setSearchTerm(''), 0);
    }
  };
```

younghyunlee22 replied:
Enter 키 사용해도 trim 잘 됩니다. 
```
  const handleSearch = () => {
    const trimmedSearchTerm = searchTerm.trim();
    if (trimmedSearchTerm === '') {
      alert('Please enter a search term.');
      return;
    }
    const searchCondition = encodeURIComponent(JSON.stringify({ searchTerm: trimmedSearchTerm, page: 1, pageSize: 20 }));
    router.push(`/search/result?searchCondition=${searchCondition}`);
    dispatch(getBooksSearchRequest({ page: initialPage, pageSize: initialPageSize, searchTerm: trimmedSearchTerm }));
    setSearchTerm('');
  };

  const handleKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
    if (event.key === 'Enter') {
      handleSearch(); // Enter 키를 눌렀을 때 handleSearch 호출
    }
  };
```
  
  엔터키 누르면 handleSearch 호출하고 그 안에서 trim이 일어나고 있어요~

younghyunlee22 committed:
fix: delay setSearchTerm reset with setTimeout

- Prevent searchTerm from being reset too early by using setTimeout.
- Fixes an issue where only part of the search term was being used in dispatch due to premature state reset.
